### PR TITLE
Added GnatMQ broker

### PR DESCRIPTION
**Pull request by VladimirAkopyan**
https://github.com/gnatmq/gnatmq

GnatMQ broker is awesome because it allows embedding a full broker in cloud servers or UWP apps. It works great for IoT projects.

### CODE REVIEW

**Comment by thangchung** [comment]
@VladimirAkopyan Is it working with .NET Core? I have seen that `<TargetFrameworkVersion>v4.5</TargetFrameworkVersion>` in the *.csproj. Could you please clarify a bit?

**Comment by VladimirAkopyan** [comment]
Hey @thangchung , add https://github.com/chkr1011/MQTTnet instead, that's MQTT broker that does essentially the same thing, but is more modern, popular and actively developed. 

GnatMQ does support core, it just isn't on the master branch, https://github.com/gnatmq/gnatmq/tree/IntegrationAPI/GnatMQ. However GnatMQ developed is halted at this point.

**Comment by ngohungphuc** [comment]
@VladimirAkopyan so this project not support asp.net core ?

File: README.md [hunk 1]
```diff
@@ -338,6 +338,7 @@ Thanks to all [contributors](https://github.com/thangchung/awesome-dotnet-core/g
 * [emitter](https://emitter.io/) - Free open source real-time messaging service that connects all devices. This publish-subscribe messaging API is built for speed and security.
 * [EventStore](https://github.com/EventStore/EventStore) - The open-source, functional database with Complex Event Processing in JavaScript. [https://geteventstore.com](https://geteventstore.com)
 * [Foundatio](https://github.com/exceptionless/Foundatio#queues) - A common interface with in memory, redis and azure implementations.
+* [GnatMQ](https://github.com/gnatmq/gnatmq) - Embedded MQTT broker/server, open-source and 100% implemented in .Net
```
Comment (ngohungphuc) [inline README.md]: @VladimirAkopyan so this project not support asp.net core ?

Comment (VladimirAkopyan) [reply README.md]: It does support .Net Core, .Net Standard, Linux and UWP
It's not related to ASP or web things

Comment (thangchung) [reply README.md]: @ngohungphuc I just have a look at https://github.com/gnatmq/gnatmq/tree/IntegrationAPI/GnatMQ. Actually, they are working for .NET Core version there. I think we just keep it `OPEN` until they merge it to `master` branch?

Comment (ngohungphuc) [reply README.md]: Agree

**Pull request closed without merging**